pkg/api/types.go

```diff
@@ -573,6 +573,9 @@ type MasterProfile struct {
 	FQDN string `json:"fqdn,omitempty"`
 	// True: uses cosmos etcd endpoint instead of installing etcd on masters
 	CosmosEtcd *bool `json:"cosmosEtcd,omitempty"`
+
+	// Enable an alternative SSH port 54322 to prevent the default port 22 being blocked
+	SSHAlternativePort bool `json:"sshAlternativePort,omitempty"`
```

Comment:
would it make sense to make this a configurable number instead to allow supporting more SSH ports in the future?

Reply:
The `/etc/ssh/sshd_config` has to be modified and I change it by extension. Since I'm not familiar with aks-engine I have no idea how to add a user-defined parameter to sshd_config. Is there a way?

Reply:
Instead of using an extension to modify `/etc/ssh/sshd_config`, you could use cloud init or CSE (cse_config.sh is probably the bast place to put this) to write to that file during cluster provisioning, and pass in an environment variable as the value of the port.

So I would add something like this to cse_config.sh (and call the new func from cse_main.sh):

```
configureSSHAlternativePort() {
    if [[ -z "${SSH_ALTERNATIVE_PORT}" ]]; then
        sed -i "s/^Port 22$/Port 22\nPort ${SSH_ALTERNATIVE_PORT}/1" /etc/ssh/sshd_config
        /etc/init.d/ssh restart
    fi
}
```

Does that make sense? I'm also happy to do a pair programming session with you if that helps :)

Reply:
Thank you so much @CecileRobertMichon I use cloud init and it works.

Reply:
looks good, added some comments